### DISCUSSION TITLE
关于 Label 和 TextBlock 的文本内容格式化问题

### DISCUSSION THREAD

**Discussion by KiwiR1der** [BYJRK/DotNet-Discussions#123] [WPF 开发]
今天在使用`Label`绑定`DateTime`时间的时候产生的疑问，为什么`Label`的`Content`的`StringFormat`不会生效呢，换成`TextBlock`是可以的：
![image](https://github.com/BYJRK/DotNet-Discussions/assets/73887119/2d4218e9-f635-43ba-9d27-dbbd16bfd6aa)

**Comment by BYJRK** [comment]
`Label` 要用 `ContentStringFormat` 属性。`StringFormat` 只对 string 类型的属性有效

```xaml
<TextBlock Text="{Binding Text, StringFormat='The day is {0}'}" />
<Label Content="{Binding Text}" ContentStringFormat="The day is {0}" />
```
